File: app/src/main/java/com/kaciula/archiman/data/util/gson/GsonLocalDateTimeAdapter.java
package com.kaciula.archiman.data.util.gson;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonToken;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;
import org.joda.time.LocalDateTime;
import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

public class GsonLocalDateTimeAdapter extends TypeAdapter<LocalDateTime> {

  private static final String DATE_TIME_FORMAT = "yyyy-MM-dd HH:mm:ss";
  private static final DateTimeFormatter DATE_TIME_FORMATTER =
      DateTimeFormat.forPattern(DATE_TIME_FORMAT);

  @Override
  public void write(JsonWriter writer, LocalDateTime value) throws IOException {
    if (value != null) {
      writer.value(value.toString(DATE_TIME_FORMAT));
    } else {
      writer.nullValue();
    }
  }

  @Override
  public LocalDateTime read(JsonReader reader) throws IOException {
    if (reader.peek() == JsonToken.NULL) {
      reader.nextNull();
      return null;
    }
    String result = reader.nextString();
    return DATE_TIME_FORMATTER.parseLocalDateTime(result);
  }
}

File: app/src/main/java/com/kaciula/archiman/data/util/gson/GsonLocalDateTimeAdapter.kt
package com.kaciula.archiman.data.util.gson

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonToken
import com.google.gson.stream.JsonWriter
import org.joda.time.LocalDateTime
import org.joda.time.format.DateTimeFormat
import java.io.IOException

class GsonLocalDateTimeAdapter : TypeAdapter<LocalDateTime>() {

    @Throws(IOException::class)
    override fun write(writer: JsonWriter, value: LocalDateTime?) {
        if (value != null) {
            writer.value(value.toString(DATE_TIME_FORMAT))
        } else {
            writer.nullValue()
        }
    }

    @Throws(IOException::class)
    override fun read(reader: JsonReader): LocalDateTime? {
        if (reader.peek() == JsonToken.NULL) {
            reader.nextNull()
            return null
        }
        val result = reader.nextString()
        return DATE_TIME_FORMATTER.parseLocalDateTime(result)
    }

    companion object {

        private val DATE_TIME_FORMAT = "yyyy-MM-dd HH:mm:ss"
        private val DATE_TIME_FORMATTER = DateTimeFormat.forPattern(DATE_TIME_FORMAT)
    }
}
